libcontainer/console.go

```diff
@@ -11,5 +16,59 @@ type Console interface {
 	Path() string
 
 	// Fd returns the fd for the master of the pty.
-	Fd() uintptr
+	File() *os.File
+}
+
+const (
+	TerminalInfoVersion uint32 = 201610041
+	TerminalInfoType    uint8  = 'T'
+)
+
+// TerminalInfo is the structure which is passed as the non-ancillary data
+// in the sendmsg(2) call when runc is run with --console-socket. It
+// contains some information about the container which the console master fd
+// relates to (to allow for consumers to use a single unix socket to handle
+// multiple containers). This structure will probably move to runtime-spec
+// at some point. But for now it lies in libcontainer.
+type TerminalInfo struct {
+	// Version of the API.
+	Version uint32 `json:"version"`
+
+	// Type of message (future proofing).
+	Type uint8 `json:"type"`
```

Comment:
I changed it from a string to a `uint8` (I don't remember who suggested the change). I don't really want to switch back and forth over _one field_ in a structure for a ~2k line patch. 

Put simply this is why I went with having a type field: It means that we can future proof the specification of types. Now, if you want to write protocol version negotiation code for `--console-socket` you can go ahead. But personally I think that's massively overkill. We're just sending a **single** file descriptor. If we want to expand this later we can expand it inside the OCI `runtime-spec` (where this structure will end up living).